swift/extractor/translators/ExprTranslator.cpp

```diff
@@ -671,4 +671,11 @@ codeql::MaterializePackExpr ExprTranslator::translateMaterializePackExpr(
   return entry;
 }
 
+codeql::ExtractFunctionIsolationExpr ExprTranslator::translateExtractFunctionIsolationExpr(
+    const swift::ExtractFunctionIsolationExpr& expr) {
+  auto entry = createExprEntry(expr);
+  entry.function_expr = dispatcher.fetchLabel(expr.getFunctionExpr());
```

Comment:
Is this line something that needs to be added for each entry representing an expression, so we know the function from which it is coming? Also, can expressions only occur in a function context, or is this specific to this particular expression?

Reply:
Thanks. So this basically you looking at the AST node of `ExtractFunctionIsolationExpr` that is coming from the Swift frontend, and seeing that it has a `getFunctionExpr` method which is relevant?

Reply:
exactly. Other sources of information are the doc comments of the corresponding entity in the swift headers, doing a git blame on https://github.com/apple/swift and look at the PR introducing the new entity, a look at the release notes of the new version. When one gets to produce the new entity in the generated test directory that's the best scenario, as it allows to double check everything's in place: this was the case for this specific node, and is in general the case for actual new syntactic entities.

But alas it's not always that easy, because we're seeing entities at a deeper stage than normal AST parsing, and it's sometimes hard to reconstruct when the compiler inserts a certain synthesized node.